### PR TITLE
Refactor exporter creation functions

### LINKED ISSUE #1979: Exporter convenience functions do not set a service name by default
The `InstallNewPipeline` and `NewExportPipeline` create a `TracerProvicer` that does not have any `Resource` associated with it. This means that by default the service name, a required attribute, will not be set and the default value will be used. The purpose of these functions was to provide users with a simple one-call solution to setup exporters with all the required values included and production level configuration set. Not setting the service name means these functions are not serving their purpose.

## Possible solutions

1. Remove these functions. They can be useful, but, as this issue points out, their API may not be the most validate with user feedback. Removing these functions prior to a stable release means they can be added back in with more context from measured use.
2. Add a way for the service name to be passed.
   1. Add an argument for the service name. This will resolve this issue, but will not be extensible.
   2. Add a configuration/options that include the service name as arguments. This will solve this issue, but will be cumbersome and add API bloat.
3. Leave it as is. These functions will remain incomplete in their setup with best practices but they can be documented that you should not use them if you want your service to be identified, or that you should use environment variables.

### PR DESCRIPTION
**Pull request by MrAlias**
- Remove all `InstallNewPipeline`/`NewExportPipeline` functions from exporters. This resolves #1979 
- Rename all `NewExporter` functions to be just `New`. This removes the redundancy of the name and follows common Go naming patterns

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -74,6 +74,12 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - The `Get` method of the `TraceState` type from the `go.opentelemetry.io/otel/trace` package has been updated to accept a `string` instead of an `attribute.Key` type. (#1931)
 - The `Insert` method of the `TraceState` type from the `go.opentelemetry.io/otel/trace` package has been updated to accept a pair of `string`s instead of an `attribute.KeyValue` type. (#1931)
 - The `Delete` method of the `TraceState` type from the `go.opentelemetry.io/otel/trace` package has been updated to accept a `string` instead of an `attribute.Key` type. (#1931)
+- Rename `NewExporter` to `New` in the `go.opentelemetry.io/otel/exporters/stdout` package. (TBD)
+- Rename `NewExporter` to `New` in the `go.opentelemetry.io/otel/exporters/metric/prometheus` package. (TBD)
+- Rename `NewExporter` to `New` in the `go.opentelemetry.io/otel/exporters/trace/jaeger` package. (TBD)
+- Rename `NewExporter` to `New` in the `go.opentelemetry.io/otel/exporters/trace/zipkin` package. (TBD)
+- Rename `NewExporter` to `New` in the `go.opentelemetry.io/otel/exporters/otlp` package. (TBD)
+- Rename `NewUnstartedExporter` to `NewUnstarted` in the `go.opentelemetry.io/otel/exporters/otlp` package. (TBD)
```
Comment (MrAlias) [inline CHANGELOG.md:82]:
```suggestion
- Rename `NewExporter` to `New` in the `go.opentelemetry.io/otel/exporters/stdout` package. (#1985)
- Rename `NewExporter` to `New` in the `go.opentelemetry.io/otel/exporters/metric/prometheus` package. (#1985)
- Rename `NewExporter` to `New` in the `go.opentelemetry.io/otel/exporters/trace/jaeger` package. (#1985)
- Rename `NewExporter` to `New` in the `go.opentelemetry.io/otel/exporters/trace/zipkin` package. (#1985)
- Rename `NewExporter` to `New` in the `go.opentelemetry.io/otel/exporters/otlp` package. (#1985)
- Rename `NewUnstartedExporter` to `NewUnstarted` in the `go.opentelemetry.io/otel/exporters/otlp` package. (#1985)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -96,6 +102,8 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - The `IsEmpty` method of the `TraceState` type in the `go.opentelemetry.io/otel/trace` package is removed in favor of using the added `TraceState.Len` method. (#1931)
 - The `Set`, `Value`, `ContextWithValue`, `ContextWithoutValue`, and `ContextWithEmpty` functions in the `go.opentelemetry.io/otel/baggage` package are removed.
   Handling of baggage is now done using the added `Baggage` type and related context functions (`ContextWithBaggage`, `ContextWithoutBaggage`, and `FromContext`) in that package. (TBD)
+- The `InstallNewPipeline` and `NewExportPipeline` creation functions in all the exporters (prometheus, otlp, stdout, jaeger, and zipkin) have been removed.
+  These functions were deemed permature attempts to provide convenience that did not achieve this aim. (TBD)
```
Comment (MrAlias) [inline CHANGELOG.md:106]:
```suggestion
  These functions were deemed permature attempts to provide convenience that did not achieve this aim. (#1985)
```

File: exporters/otlp/otlptrace/otlptracegrpc/exporter.go [hunk 3]
```diff
@@ -18,27 +18,14 @@ import (
...
-// NewExporter constructs a new Exporter and starts it.
-func NewExporter(ctx context.Context, opts ...Option) (*otlptrace.Exporter, error) {
-	return otlptrace.NewExporter(ctx, NewClient(opts...))
+// New constructs a new Exporter and starts it.
+func New(ctx context.Context, opts ...Option) (*otlptrace.Exporter, error) {
+	return otlptrace.New(ctx, NewClient(opts...))
 }
 
 // NewUnstartedExporter constructs a new Exporter and does not start it.
 func NewUnstartedExporter(opts ...Option) *otlptrace.Exporter {
```
Comment (Aneurysm9) [inline exporters/otlp/otlptrace/otlptracegrpc/exporter.go:29]:
```suggestion
func NewUnstarted(opts ...Option) *otlptrace.Exporter {
```

Should this drop `Exporter` for consistency with the outgoing `otlp` exporter?

File: exporters/otlp/otlptrace/otlptracehttp/exporter.go [hunk 4]
```diff
@@ -18,27 +18,14 @@ import (
...
-// NewExporter constructs a new Exporter and starts it.
-func NewExporter(ctx context.Context, opts ...Option) (*otlptrace.Exporter, error) {
-	return otlptrace.NewExporter(ctx, NewClient(opts...))
+// New constructs a new Exporter and starts it.
+func New(ctx context.Context, opts ...Option) (*otlptrace.Exporter, error) {
+	return otlptrace.New(ctx, NewClient(opts...))
 }
 
 // NewUnstartedExporter constructs a new Exporter and does not start it.
 func NewUnstartedExporter(opts ...Option) *otlptrace.Exporter {
```
Comment (Aneurysm9) [inline exporters/otlp/otlptrace/otlptracehttp/exporter.go:29]:
```suggestion
func NewUnstarted(opts ...Option) *otlptrace.Exporter {
```

Also here.

File: exporters/otlp/otlptrace/exporter_test.go [hunk 5]
```diff
@@ -51,16 +46,3 @@ func (m *noopClient) UploadTraces(_ context.Context, _ []*tracepb.ResourceSpans)
 
 func (m *noopClient) Reset() {
```
Comment (paivagustavo) [inline exporters/otlp/otlptrace/exporter_test.go:47]: I think this test file can be removed, as it doesn't test anything now.

**Review by paivagustavo — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**